ARCHITECTURE.md

```diff
@@ -0,0 +1,71 @@
+Architecture of the library
+===
+
+    ELF -> Specifications -> Objects -> Links
+
+ELF
+---
+
+BPF is usually produced by using `clang` to compile a subset of C. `clang` outputs
```

Comment:
`clang` is a command and "Clang" is a product/tool. I think in most of your cases you should remove the backticks, except in the case of `CollectionSpec` (which should have a godoc link or something).